### PR TITLE
STORM-2305: STORM-2279 calculates task index different from grouper code

### PR DESCRIPTION
**Pull request by raghavgautam**
@HeartSaVioR @arunmahadevan Please review.

### CODE REVIEW

**Comment by HeartSaVioR** [comment]
+1

File: storm-core/src/jvm/org/apache/storm/daemon/nimbus/Nimbus.java [hunk 1]
```diff
@@ -3600,8 +3600,7 @@ public ComponentPageInfo getComponentPageInfo(String topoId, String componentId,
                 List<Integer> tasks = compToTasks.get(StormCommon.EVENTLOGGER_COMPONENT_ID);
                 tasks.sort(null);
                 // Find the task the events from this component route to.
-                int taskIndex = (TupleUtils.listHashCode(Arrays.asList(componentId)) % tasks.size() + tasks.size()) %
-                        tasks.size();
+                int taskIndex = TupleUtils.chooseTaskIndex(Arrays.asList(componentId), tasks.size());
```
Comment (arunmahadevan) [inline storm-core/src/jvm/org/apache/storm/daemon/nimbus/Nimbus.java]: nit: use `Collections.singletonList(componentId)`

File: storm-core/src/jvm/org/apache/storm/utils/TupleUtils.java [hunk 2]
```diff
@@ -40,7 +40,11 @@ public static boolean isTick(Tuple tuple) {
              && Constants.SYSTEM_TICK_STREAM_ID.equals(tuple.getSourceStreamId());
     }
 
-    public static <T> int listHashCode(List<T> alist) {
+    public static <T> int chooseTaskIndex(List<T> alist, int numTasks) {
```
Comment (arunmahadevan) [inline storm-core/src/jvm/org/apache/storm/utils/TupleUtils.java]: nit: rename alist to keys to better convey the meaning.

File: storm-core/src/jvm/org/apache/storm/daemon/nimbus/Nimbus.java (hunk 1 above)
Comment (raghavgautam) [inline storm-core/src/jvm/org/apache/storm/daemon/nimbus/Nimbus.java]: Done.

File: storm-core/src/jvm/org/apache/storm/utils/TupleUtils.java (hunk 2 above)
Comment (raghavgautam) [inline storm-core/src/jvm/org/apache/storm/utils/TupleUtils.java]: Done.

**Comment by arunmahadevan** [comment]
+1

**Pull request merged**